submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireService.kt

```diff
@@ -21,14 +18,11 @@ class FireService(
     private val client: FireClient,
     private val fireTempDirPath: File,
 ) {
-
-    fun cleanFtp(sub: ExtSubmission) {
-        logger.info { "Started cleaning FTP files (unset path) of files in submission ${sub.accNo}" }
-        client
-            .findByAccNo(sub.accNo)
-            .apply { logger.info { "Found $size files in submission ${sub.accNo}" } }
-            .forEach { client.unsetPath(it.fireOid) }
-        logger.info { "Finished cleaning FTP files (unset path) of files in submission ${sub.accNo}" }
+    fun cleanFile(file: ExtFile) {
+        if (file is FireFile) {
```

Comment:
Shouldn't we filter the files before passing them to the function or maybe ask this question before? I just think this hides the fact that it won't do anything if the file is NfsFile so if I see a method that gets an ExtFile called cleanFile I'd expect it would clean both types